### ISSUE TITLE
Respec breaks the Erdof fight bug

### ISSUE THREAD

**Issue by ChrisPJohn** [ChrisPJohn/SolastaCommunityExpansion#127]
goofsterstud
The "Respect" Edrof fight bug (game breaking) is back.

I am running into the same bug on the forms. Apparently the old respec was the reason. I have respeced each of my characters and now can not finish. You can read more about what I'm going through here:https://forums.solasta-game.com/forum/stuck-in-fight-with-edrofThe original Respec mod had a bunch of posts on this same thing.

---------------------

My notes: I haven't reproed this or seen logs yet for it so this is currently a complete guess, but I wonder if there's some initialization for a character that happens when they're added to the campaign that is missing during respec. In GameManager::CreateGameAsync there's several steps used while loading a character in, I'll need to investigate more to see which ones may be covered by respec.

**Comment by ThyWoof** [comment]
this bug was fixed about 2-3 months ago. unfortunately it won't fix any save that respec'ed before the fix.

**Comment by ThyWoof** [comment]
it was caused as I didn't reuse the hero GUID on the respec'ed causing this particular quest to fail on sync.

**Comment by ChrisPJohn** [comment]
goofsterstud
Community Expansion. I never used the original.

I'm unable to get you the saves...since Solasta limits you to 50, I used them. I was able to forfeit the fight (unfortunate). After that the game decided I couldn't equip armor (I could before I had to forfeit). Then I respecced again and could fit armor.

Would a save way after it happened even help?  This is a great addon. Highly recommend it, even with the bug. Guess I shouldn't respec everything.

---------------------
It sound like there is still something going on with respec and that fight. Given the comment about being unable to equip armor after the fight I wonder if we need to call Register on the new hero that's created. Register handles a lot of item/inventory bookkeeping.
